src/commonMain/kotlin/org/mongodb/kbson/serialization/Serializers.kt

```diff
@@ -68,11 +74,168 @@ import org.mongodb.kbson.BsonTimestamp
+ *
+ * val ejson = Ejson
+ * val instance = DataHolder(42, "some data", BsonObjectId() }
+ *
+ * // Plain StringFormat usage
+ * val stringOutput: String = ejson.encodeToString(instance)
+ *
+ * // BsonValue serialization
+ * val bsonValue: BsonValue = ejson.encodeToBsonValue(instance)
+ *
+ * // Deserialize from string
+ * val deserialized: DataHolder = ejson.decodeFromString<DataHolder>(stringOutput)
+ *
+ * // Deserialize from BsonValue
+ * val deserializedFromBsonValue: DataHolder = ejson.decodeFromBsonValue<DataHolder>(bsonValue)
+ *
+ *  // Deserialize from string to BsonValue
+ *  val deserializedToBsonValue: BsonValue = Bson(stringOutput)
+ * ```
+ *
+ * It does not support polymorphic serializers yet.
+ */
+@ExperimentalApi
+public sealed class Ejson(
+    public val ignoreUnknownKeys: Boolean,
+    private val json: Json
+) : StringFormat {
+
+    /**
+     * The default instance of [Ejson] with default configuration.
+     */
+    public companion object Default : Ejson(
+        ignoreUnknownKeys = true,
+        json = Json
+    )
+
+    override val serializersModule: SerializersModule = json.serializersModule
+
+    /**
+     * Serializes the given [value] into an equivalent [BsonValue] using the given [serializer]
+     *
+     * @throws [SerializationException] if the given value cannot be serialized to EJSON
+     */
+    public fun <T> encodeToBsonValue(
+        serializer: SerializationStrategy<T>,
+        value: T
+    ): BsonValue = writeBson(value, serializer)
+
+    /**
+     * Deserializes the given [value] into a value of type [T] using the given [deserializer].
+     *
+     * @throws [SerializationException] if the given EJSON element is not a valid EJSON input for the type [T]
+     */
+    public fun <T> decodeFromBsonValue(
+        deserializer: DeserializationStrategy<T>,
+        value: BsonValue
+    ): T = readBson(value, deserializer)
+
+    /**
+     * Deserializes the given EJSON [string] into a value of type [T] using the given [deserializer].
+     *
+     * @throws [SerializationException] if the given EJSON string is not a valid EJSON input for the type [T]
+     */
+    public override fun <T> decodeFromString(
+        deserializer: DeserializationStrategy<T>,
+        string: String
+    ): T = decodeFromBsonValue(
+        deserializer,
+        json.decodeFromString(string)
+    )
+
+    /**
+     * Serializes the [value] into an equivalent EJSON using the given [serializer].
+     *
+     * @throws [SerializationException] if the given value cannot be serialized to EJSON.
+     */
+    public override fun <T> encodeToString(
+        serializer: SerializationStrategy<T>,
+        value: T
+    ): String = json.encodeToString(
+        encodeToBsonValue(
+            serializer,
+            value
+        )
+    )
+}
+
+/**
+ * Creates an instance of [Ejson] configured with a [ignoreUnknownKeys] and a custom [serializersModule].
+ */
+@ExperimentalApi
+public fun Ejson(
```

Comment:
NamingIsHard: This is "Extended JSON" should this be `EJson` instead?

Reply:
Either one sounds good to me, gonna change it to `EJson`